### PR TITLE
require deposit to be divisible by GWEI

### PR DESCRIPTION
**Pull request by MrChico**

### CODE REVIEW

File: deposit_contract.sol [hunk 1]
```diff
@@ -80,8 +79,10 @@ contract DepositContract is IDepositContract {
         require(deposit_count < MAX_DEPOSIT_COUNT);
 
         // Check deposit amount
+        require(msg.value >= 1 ether);
+        require(msg.value % GWEI == 0, "DepositContract: deposit value not a multiple of gwei");
```
Comment (axic) [inline deposit_contract.sol]: Can we keep all the revert strings in the #11 PR?

Also in this one the first `require` is missing the string.

Comment (MrChico) [reply deposit_contract.sol]: well, we need to decide on an ordering of these PRs first. This one assumes `#11` is merged first. But you are right that the above one is inconsistent with that assumption

File: deposit_contract.sol [hunk 2]
```diff
@@ -80,8 +79,10 @@ contract DepositContract is IDepositContract {
         require(deposit_count < MAX_DEPOSIT_COUNT);
 
         // Check deposit amount
+        require(msg.value >= 1 ether);
```
Comment (axic) [inline deposit_contract.sol]: I think it would still make sense keeping `MIN_DEPOSIT_AMOUNT = 1 ether` as a const?

Comment (MrChico) [reply deposit_contract.sol]: I don't like the indirection of these constants tbh

Comment (axic) [reply deposit_contract.sol]: How about we keep the constants and clear it up a separate PR?

I think we could
- remove the `_LENGTH` constants after #10
- remove `MAX_DEPOSIT_COUNT` and making it part the require statement prior to typecasting 
- and also drop the `MIN_DEPOSIT_AMOUNT`

The only two constants which make sense are `GWEI` (too late to get https://github.com/ethereum/solidity/issues/8716 done) and `DEPOSIT_CONTRACT_TREE_DEPTH`.

Comment (MrChico) [reply deposit_contract.sol]: Yes. That sounds like a good idea. I'd maybe keep `MAX_DEPOSIT_COUNT`, but let's take that when we get there

**Review by axic — APPROVED** [review]

**Pull request merged**